### DISCUSSION TITLE
"WindiCSS for Newbies" Needed

### DISCUSSION THREAD

**Discussion by iocouto** [windicss/windicss#532] [Ideas]
The current docs seem to be written for a very specific audience, who already...:
- ...have extensive experience with Javascript, and possibly Typescript
- ...are comfortable working with various pre- and post-processing tools - such as Vite, Rollup, etc.
- ...regularly use large Javascript frameworks and libraries - such as Vue and React
- ...are quite familiar with TailwindCSS

As a consequence, I have found the entry requirements to WindiCSS quite strict. If, for example, I am a Frontend Developer used to working with traditional frameworks like Bootstrap, Foundation, UIKit, etc., and using GUI-based compilation tools - like CodeKit and PrePros - then the documentation can overwhelm rather than help. For coders that are new to web development, and have more familiarity with HTML and CSS than JS (and its confusing ecosystem), the current WindiCSS docs can make it seem terribly technical and unreachable. And the irony here is, that those entry-level coders are among those who would benefit the most from using Windi.

I'd like to suggest, therefore, that WindiCSS should have a "Start Here!" or "WindiCSS for Newbies" set of docs, that actually assumes very little knowledge, and takes new users by the hand, step-by-step through the process of installing, configuring and using WindiCSS - and do not assume familiarity with TailwindCSS, or any other framework, library or tool.

Here is an example for an 'Essential WindiCSS' docs outline:

- What is WindiCSS
- Installing WindiCSS
    - Requirements
    - Installing Node+npm (links to external installation docs)
    - Installing WindiCSS (to use from CLI)
    - Setting up 'dev' and 'build' npm scripts
- First Steps: setup a simple HTML page, and style elements using built-in WindiCSS styles
    - Page Layout: using container, flexbox and grid 
    - Positioning, Sizing & Spacing: using width, height, margin, padding and positioning
    - Colours: using background colours, text colours and border colours - and how to define our own custom colours
    - Advanced colours: using transparency and gradients
    - Type: how to use our own fonts, and work with font sizes, as well paragraph and text styling classes
- Adding Interactivity
    - Using hover, focus, and other interaction variants 
    - Adding transitions, delays and built-in animations 
    - Defining custom animations
- Responsiveness
    - Understanding screen variants and the 'mobile-first' approach
    - Handling responsiveness exceptions using "<" and "@" variants
- Reusing Styles
    - Defining often-used styles using config shortcuts
    - Defining often-used styles using @apply directive
- Extending WindiCSS
    - Using built-in plugins - forms, typography, etc.
    - Integrating with external frameworks & libraries - eg., Vue, React, etc.

**Comment by Doomd** [comment]
Here are some more suggestions with regard to onboarding and documentation:

- It would be much more helpful if all config options were under a "Configuration" category. Every config option should have a detailed explanation AND an example of how a particular config option would affect output or implementation in code/markup.
- There is NO easy way to find COLOR documentation. The "[Colors](https://windicss.org/utilities/colors.html)" page literally is just a list of colors. How does a user implement those colors? What are all the class properties that colors can be added too? Colors should probably have it's own heading as well. How _exactly_ should one add customizable colors to the config file? And how exactly would someone implement these customizations in their markup? There needs to be a simple walkthrough that shows how a user can rename a colorset...and how they can create global "theme" colors. The walkthrough should include the configuration file changes AND how to implement these changes in their markup. I'm a professional coder and I still haven't fully figured this out.
- I suggest renaming "Utilities" in the top menu to "Documentation". Then make sure that the documentation is exhaustive and complete.
- I have only followed the Sveltekit instructions myself, and it's very confusing to have two different sets of instructions (with vite and without Vite) without very explicitly mentioning the two different install paths. It doesn't help that Svelte (not-Sveltekit) is lumped into the same section. Every set of different instructions should have it's own separate section/heading. I only got Windi working on Sveltekit after about an hour with some trial and error (with the vite plugin).  These instructions need to be clear...and they need to be checked that they work from scratch by an actual human being...and updated when they need to be.

Those are just a few off the top of my head. Bottom line, I'm a big fan of windicss and I want to see it become popular...but I think the documentation pales in comparison to the documentation on tailwindcss.com and I absolutely agree that poor documentation will hobble a framework's success. That being said, I'm willing to help...

**Comment by jakecodev** [comment]
Agreed. Having trouble using directives in cli . I would say Windi has amazing features for experienced users but still need to consider basic users too. 

Those who have got used to Tailwind might reconsider before trying Windi, but those newbies like me who just step into atomic styling really need more detailed doc and examples. Wish Windi will become more friendly to normal users.
